src/main/java/eu/dissco/core/digitalspecimenprocessor/service/HandleService.java

```diff
@@ -30,62 +64,261 @@
+    fdoRecord.add(
+        new HandleAttribute(FIELD_IDX.get(REFERENT_DOI_NAME), REFERENT_DOI_NAME, handle.getBytes(
+            StandardCharsets.UTF_8)));
+
+    // 43: primaryReferentType
+    fdoRecord.add(
+        new HandleAttribute(FIELD_IDX.get(PRIMARY_REFERENT_TYPE), PRIMARY_REFERENT_TYPE,
+            "creation".getBytes(StandardCharsets.UTF_8)));
+
+    // 44: referent
+    fdoRecord.add(new HandleAttribute(FIELD_IDX.get(REFERENT), REFERENT,
+        TO_BE_FIXED.getBytes(StandardCharsets.UTF_8)));
+
+    // 210: objectType
+    fdoRecord.add(
+        new HandleAttribute(FIELD_IDX.get(OBJECT_TYPE), OBJECT_TYPE,
+            "Digital Specimen".getBytes(StandardCharsets.UTF_8)));
+
+    return fdoRecord;
+  }
+
+  private List<HandleAttribute> fillFdoRecordSpecimenAttributes(DigitalSpecimen digitalSpecimen) {
+    List<HandleAttribute> fdoRecord = new ArrayList<>();
+    // 42: referentName
+    fdoRecord.add(
+        new HandleAttribute(FIELD_IDX.get(REFERENT_NAME), REFERENT_NAME,
+            getAttributeFromDigitalSpecimen(digitalSpecimen, ODS_PREFIX + "specimenName",
+                UNKNOWN).getBytes(StandardCharsets.UTF_8)));
+
+    // 200: SpecimenHost
+    fdoRecord.add(
+        new HandleAttribute(FIELD_IDX.get(SPECIMEN_HOST),
+            SPECIMEN_HOST,
+            getAttributeFromDigitalSpecimen(digitalSpecimen, ODS_PREFIX + "organisationId",
+                UNKNOWN).getBytes(StandardCharsets.UTF_8)));
+
+    // 201: Specimen Host Name
+    fdoRecord.add(
+        new HandleAttribute(FIELD_IDX.get(SPECIMEN_HOST_NAME), SPECIMEN_HOST_NAME,
+            getAttributeFromDigitalSpecimen(digitalSpecimen, ODS_PREFIX + "organisationName",
+                UNKNOWN).getBytes(StandardCharsets.UTF_8)));
+
+    // 202: PrimarySpecimenObjectId
+    fdoRecord.add(
+        new HandleAttribute(FIELD_IDX.get(PRIMARY_SPECIMEN_OBJECT_ID),
+            PRIMARY_SPECIMEN_OBJECT_ID,
+            digitalSpecimen.physicalSpecimenId().getBytes(
+                StandardCharsets.UTF_8)));
+
+    // 203: primarySpecimenObjectIdType
+    fdoRecord.add(
+        new HandleAttribute(FIELD_IDX.get(PRIMARY_SPECIMEN_OBJECT_ID_TYPE),
+            PRIMARY_SPECIMEN_OBJECT_ID_TYPE,
+            getAttributeFromDigitalSpecimen(digitalSpecimen, ODS_PREFIX + "physicalSpecimenIdType",
+                UNKNOWN).getBytes(StandardCharsets.UTF_8)));
+
+    // 204: primarySpecimenObjectIdName
+    var collectionId = getAttributeFromDigitalSpecimen(digitalSpecimen,
+        ODS_PREFIX + "physicalSpecimenCollection", "");
+    if (!collectionId.isEmpty()) {
+      var idName = ("Local identifier for collection " + collectionId).getBytes(
+          StandardCharsets.UTF_8);
+      fdoRecord.add(new HandleAttribute(FIELD_IDX.get(PRIMARY_SPECIMEN_OBJECT_ID_NAME),
+          PRIMARY_SPECIMEN_OBJECT_ID_NAME, idName));
+    }
+
+    // 206: otherSpecimenIds
+    var otherSpecimenIds = getAttributeFromDigitalSpecimen(digitalSpecimen, "dwca:id", "");
```

Comment:
This will probably change, but think this is fine for now. We need to do something with the harmonisation for the other Ids.

Reply:
[Jira](https://naturalis.atlassian.net/browse/DD-454)